### ISSUE TITLE
cassandra-stress doesn't support UDT in user-profile mode

### ISSUE THREAD

**Issue by yarongilor** [scylladb/scylla-tools-java#332]
Creating a UDT and using it in a table schema is not supported by cassandra-stress user-profile mode.
This ability is required for testing specific user-profiles, similar to what is used by customers.

A Dtest for that failed like:
Using a user-profile table schema of:
```
table_definition: |
  CREATE TABLE IF NOT EXISTS user_with_ck (
    key blob,
    // Case sensitive column names
    //"Md5" text,
    md5 text,
    ...
    hash_bashname hash_and_basename,
    PRIMARY KEY (key, md5)
  ); 
```

The following code:
```
        with self.patient_cql_connection(node) as session:
            create_ks(session=session, name='keyspace_complex', rf=3)
            session.execute("CREATE TYPE IF NOT EXISTS hash_and_basename (md5 text, basename text)")
            wait_for_schema_agreement(session)
        time.sleep(30)
        logger.debug('Running stress ...')
        node.stress(["user", "no-warmup", "profile=%s" % os.path.realpath('test_data/batch-test/complex_schema.yaml'),
        ...
```
Failed with:
```
E           stderr: java.lang.UnsupportedOperationException: Because of this name: hash_bashname if you removed it from the yaml and are still seeing this, make sure to drop table
```

It got a similar failure testing in SCT:
```
< t:2023-06-01 14:03:34,429 f:common.py       l:1742 c:utils                p:DEBUG > Executing CQL 'CREATE TYPE IF NOT EXISTS custom_d1.post_speaker_item (id3 bigint, name3 text, reaction_type3 tinyint);' ...
< t:2023-06-01 14:03:34,434 f:thread.py       l:52   c:cassandra.cluster    p:DEBUG > Refreshing schema in response to schema change. {'target_type': 'TYPE', 'change_type': 'CREATED', 'keyspace': 'custom_d1', 'type': 'post_speaker_item'}
Stress command completed with bad status 1: WARNING: Table 'posts' has column 'speaker_map' of unsupported type
```

**Comment by mykaul** [comment]
Is that the right repo to report this?

**Comment by mykaul** [comment]
I don't see who's going to develop it. I reckon upstream don't have it either.

**Comment by yarongilor** [comment]
ok, thanks, i see.
@fgelcer , @fruch , should we switch back to the previously suggested option of using `tlp-stress` instead?

**Comment by fruch** [comment]
Yaron, 

can you share the whole user profile file ?

**Comment by fruch** [comment]
I would recommend testing the upstream cassandra-stress from 4.1, if it's kind of working there, backporting might be easy.

Getting familiar with tlp-stress might also be a good path to start building the exact cases we need.

**Comment by fruch** [comment]
I would recommend testing the upstream cassandra-stress from 4.1, if it's kind of working there, backporting might be easy.

Getting familiar with tlp-stress might also be a good path to start building the exact cases we need.

**Comment by yarongilor** [comment]
> I would recommend testing the upstream cassandra-stress from 4.1, if it's kind of working there, backporting might be easy.
> 
> Getting familiar with tlp-stress might also be a good path to start building the exact cases we need.

@fruch , i couldn't find any updates for this limitation in cassandra repo.
I noticed a quite similar issue, opened for years by @dkropachev - https://issues.apache.org/jira/browse/CASSANDRA-15598
Dmitry can you please confirm that? do yo think we can somehow use the patch you sent so it will resolve this limitation of UDT as well?

**Comment by fruch** [comment]
https://issues.apache.org/jira/browse/CASSANDRA-15598 isn't abot UDT, it's about nested sets.

**Comment by dkropachev** [comment]
@roydahan , you are correct, i have 3 PRs to their repo, non of them endup even looked at.

@yarongilor , the error message that you see is result of having that patch, otherwise you would see it failing by panic.
There is bug is `c-s` architecture value generation logic that does not allow it to be used for nested structures, including UDTs.
Unfortunately fixing this issue would cause major changes to be made that would make our version to diverge too far from the upstream. 
So, easiest solution was to just make `c-s` to silently skip these columns, which was implemented at apache repo as https://github.com/apache/cassandra/pull/451 and at our repo as https://github.com/scylladb/scylla-tools-java/pull/144.
#144 - was merged, while #451 - recently was closed due to the 2y inactivity.

I am going to check whether this issue is sorted out on the upstream at `4.x` branch and update this issue

**Comment by roydahan** [comment]
@dkropachev assuming we are ok with the diverage between the two, how hard would it be to support the nested structures?

**Comment by dkropachev** [comment]
@roydahan , roughly 3 days of work

**Comment by roydahan** [comment]
@dkropachev do you think you'll be able to fit this task sometime soon?

**Comment by dkropachev** [comment]
@roydahan , I am sorry, next two weeks I am booked beyond capacity, I have it in mind and will find time for it after I finish what I currently have on me. Sorry for delaying it.

**Comment by roydahan** [comment]
Thanks @dkropachev, it may become high priority for us at the moment, I'll check our options around that.

**Comment by avelanarius** [comment]
@muzarski is going to look into it today. We're not sure if some quick fix is possible, but hopefully @muzarski will determine it by the end of the day.
